### PR TITLE
highlight ranges

### PR DESCRIPTION
**Pull request by ploomiz**
## Why
These changes allow for highlighting article annotations in the browser extension - see https://github.com/Notist/browser-extension/pull/5 which implements highlighting. Annotation objects must keep track of the selected DOM that corresponds the annotation's articleText. This way, we know which parts of the DOM to highlight on the frontend.
This PR makes each annotation store a `ranges` field, which indicates which part of the article's DOM to highlight in order to highlight the an annotation's article text.
e.g. A ranges field might look like this:
```
  ranges: [
    {
      start: "/p[69]/span/span",           # to start element
      end: "/p[70]/span/span",             # end element
      startOffset: 0,                      # character offset within start element
      endOffset: 120                       # character offset within end element
    }
  ]
```
And, on the frontend, we highlight the ranges to make the article text appear highlighted.

## Changes
- Added a `ranges` field to the `Annotation` schema
- Changed the `createAnnotation` method to accept the range field for top-level annotations

### CODE REVIEW

**Comment by ploomiz** [comment]
[Integrate Annotator - http://annotatorjs.org/ - with browser extension to allow highlighting on page](https://trello.com/c/VcMak841/2-integrate-annotator-http-annotatorjs-org-with-browser-extension-to-allow-highlighting-on-page)

File: server/controllers/annotation_controller.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ export const createAnnotation = (user, body, articleId) => {
       });
   } else {
```
Comment (ethanbb) [inline server/controllers/annotation_controller.js:47]: Should we also give each child annotation a ranges field, which would have the same value as its parent's? Or did you only want to store this for top-level annotations?

If we are never planning to access an annotation without also showing all of its ancestors along with the article text, then I guess it makes sense to only store the ranges in the top annotation.

File: server/models/annotation.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ const annotationSchema = new Schema({
 
   text: { type: String, trim: true },
   articleText: String,
+  ranges: { type: Array, default: [] },
```
Comment (ethanbb) [inline server/models/annotation.js]: We should be able to specify this part of the schema more specifically (assuming each member of the array should have the same format). I can change this and make a commit.

**Review by ethanbb — CHANGES_REQUESTED** [review]

File: server/controllers/annotation_controller.js (hunk 1 above)
Comment (ploomiz) [reply server/controllers/annotation_controller.js:47]: We only want to store the ranges field for top-level annotations.

File: server/models/annotation.js (hunk 2 above)
Comment (ploomiz) [reply server/models/annotation.js]: Yes, it should be more specific. Maybe something like this?
```
ranges: [
  {
    start: { type: String },
    end: { type: String },
    startOffset: { type: Number },
    endOffset: { type: Number },
  }
]
```

**Comment by ethanbb** [comment]
Wait that's weird, all my tests passed...hold on

**Review by ethanbb — APPROVED** [review]

**Pull request merged**